clouddriver-titus/src/main/groovy/com/netflix/spinnaker/clouddriver/titus/deploy/handlers/TitusDeployHandler.groovy

```diff
@@ -313,22 +313,25 @@ class TitusDeployHandler implements DeployHandler<TitusDeployDescription> {
       submitJobRequest = submitJobRequest.withDockerImageVersion(dockerImage.imageVersion)
     }
 
-    if (description.hardConstraints) {
-      description.hardConstraints.each { constraint ->
-        submitJobRequest.withConstraint(SubmitJobRequest.Constraint.hard(constraint))
-      }
-    }
-
-    if (description.softConstraints) {
-      description.softConstraints.each { constraint ->
-        submitJobRequest.withConstraint(SubmitJobRequest.Constraint.soft(constraint))
-      }
-    }
-
-    if (description.jobType == "service" && !description.hardConstraints?.contains(SubmitJobRequest.Constraint.ZONE_BALANCE) && !description.softConstraints?.contains(SubmitJobRequest.Constraint.ZONE_BALANCE)) {
-      submitJobRequest.withConstraint(SubmitJobRequest.Constraint.soft(SubmitJobRequest.Constraint.ZONE_BALANCE))
+    // constraints map takes precedence when specified in the request
```

Comment:
this is a bit convoluted (not your fault - obv titus requirements) but maybe you can add a more descriptive comment that explains what's up

Reply:
Agree , added a comment for now , I think i will follow up this with what @ajordens suggested with a preprocessor , that will make this better i think